### PR TITLE
For #9505: Adds possibility to sort saved logins

### PR DESCRIPTION
**Pull request by ValentinTimisica**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests.
- [x] **Screenshots**: This PR includes screenshots and a [video](https://drive.google.com/open?id=1Lg75RIVXjSPeDIiWiOldYqKrwJPKis-W) of the changes made.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).

<table style="width:100%">
  <tr>
    <th>Light theme</th>
    <th>Dark theme</th>
  </tr>
  <tr>
    <td></td>
    <td></td>
  </tr>
</table>

<table style="width:100%">
  <tr>
    <th>Private mode</th>
  </tr>
  <tr>
    <td></td>
  </tr>
</table>

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by ValentinTimisica** [comment]
Note that it should be merged after [this AC PR](https://github.com/mozilla-mobile/android-components/pull/6559) is merged and available in Fenix.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1200,6 +1200,12 @@
     <string name="preference_accessibility_force_enable_zoom">Zoom on all websites</string>
     <!-- Summary for Accessibility Force Enable Zoom Preference -->
     <string name="preference_accessibility_force_enable_zoom_summary">Enable to allow pinch and zoom, even on websites that prevent this gesture.</string>
+    <!-- Saved logins sorting strategy menu item -by name- (if selected, it will sort saved logins alphabetically) -->
+    <string name="saved_logins_sort_strategy_alphabetically">Name (A-Z)</string>
```
Comment (ekager) [inline app/src/main/res/values/strings.xml:1204]: I'm not sure if you'll know this, but do all languages have the concept of (A-Z)? If not, I'm assuming the translators will deal with this? Might be good to double check

Comment (ValentinTimisica) [reply app/src/main/res/values/strings.xml:1204]: I used the string provided in the [mocks](https://app.abstract.com/share/33afd8de-f90e-4504-90e3-ee2a3bcb1925?mode=design&sha=0150fa57c67a9ae5786541d50682351760e767ca). Maybe we should ask @AmyYLee or @betsymi if the string `Name (A-Z)` is ok and the translators will deal with this afterwards.

Comment (betsymi) [reply app/src/main/res/values/strings.xml:1204]: I ran this by l10n. We are already using this string in desktop, and communities have already adapted their translations. ([See Transvision](https://transvision.mozfr.org/string/?entity=browser/browser/aboutLogins.ftl:login-list-name-option&repo=gecko_strings)). What would need verification is if sorting and collation reflect the appropriate rules for the locale the user is using.

**Comment by eliserichards** [comment]
Looks like [the AC PR](https://github.com/mozilla-mobile/android-components/pull/6559) was merged! This should be ready to rebase @ValentinTimisica 
CC: @ekager

**Comment by ValentinTimisica** [comment]
Apparently it is not yet available in Fenix.

File: app/src/main/java/org/mozilla/fenix/settings/logins/SortingStrategy.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.logins
+
+sealed class SortingStrategy {
+    abstract operator fun invoke(logins: List<SavedLoginsItem>): List<SavedLoginsItem>
+
+    object Alphabetically : SortingStrategy() {
+        override fun invoke(logins: List<SavedLoginsItem>): List<SavedLoginsItem> {
+            return logins.sortedBy { it.url }
```
[the remark this answers is not in the document]
Comment (ValentinTimisica) [reply app/src/main/java/org/mozilla/fenix/settings/logins/SortingStrategy.kt:12]: Yes, good catch, I didn't initially see this problem. I will use `urlToTrimmedHost` extension to sort only by domain.

**Comment by AmyYLee** [comment]
@eliserichards Hi, I reviewed the APK and the animation of the list when it switches between "Name" and "Last Used" is a bit motion intense (the list items all move and reorder). The list items should switch to the new order without motion.

**Comment by ValentinTimisica** [comment]
Hi @AmyYLee, I removed the animation. You can check this [video](https://drive.google.com/file/d/1H8NCQ0utjtzb0YpDSaqxyQ4828WqxmQZ/view?usp=sharing) with the fix.

**Comment by AmyYLee** [comment]
> Looks great! I think this is ready to land with signoff from @AmyYLee 🔥

@ValentinTimisica @eliserichards This looks great! Approved. Thanks

**Pull request merged**